src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -197,8 +197,16 @@ public static void GetModules_GetModule(Assembly assembly)
             Assert.NotEmpty(assembly.GetModules());
             foreach (Module module in assembly.GetModules())
             {
-                Assert.NotNull(module);
-                Assert.Equal(module, assembly.GetModule(module.Name));
```

Comment:
in my experiment i see the same behavior on desktop & .net core... which is 
            Assert.NotNull(module);
            Assert.Equal(module, assembly.GetModule(module.Name));